Modules/Sources/NineAnimatorNativeSources/AnimeSources/AnimeSaturn/AnimeSaturn+Episode.swift

```diff
@@ -38,10 +41,16 @@ extension NASourceAnimeSaturn {
                         let data = responseContent
                         let utf8Text = String(data: data, encoding: .utf8) ?? String(decoding: data, as: UTF8.self)
                         let  bowl = try SwiftSoup.parse(utf8Text)
-                        let video = try bowl.select("video source").attr("src").replacingOccurrences(of: "http://", with: "https://")
+                        let video = try bowl.select("video source").attr("src")
+                        var finalurl: URL
+                        if video.isEmpty {
+                            finalurl = try URL(string: NASourceAnimeSaturn.regex.firstMatch(in: bowl.debugDescription)?.first ?? "").tryUnwrap()
```

Comment:
```suggestion
                            finalurl = try URL(string: NASourceAnimeSaturn.regex.firstMatch(in: utf8Text)?.first ?? "").tryUnwrap()
```

Don't use `.debugDescription` to obtain the original html source.